.github/workflows/publish.yml

```diff
@@ -2,36 +2,40 @@ name: Publish
 
 on:
   push:
-    tags:
-      - "v*"
+    branches-ignore:
+      # No canary deploys for branches opened by dependabot
+      - 'dependabot/**'
 
 jobs:
   publish:
-    name: Publish to NPM
+    name: Make a release and publish to NPM
 
     runs-on: ubuntu-latest
+    if: "!contains(github.event.head_commit.message, 'ci skip') && !contains(github.event.head_commit.message, 'skip ci')"
 
     steps:
       - uses: actions/checkout@v2
+
+      - name: Prepare repository
+        run: git fetch --unshallow --tags
+
       - uses: actions/setup-node@v2.4.0
         with:
           registry-url: "https://registry.npmjs.org"
           node-version: "15"
+
       - name: Install Node dependencies
         run: yarn --frozen-lockfile
+
       - name: Build
         run: yarn build
-      - name: Publish to NPM
-        run: npm publish
+
+      # npm config set scripts-prepend-node-path true
+      # don't use yarn run to avoid using yarnpkg registry
+      - name: Create release
+        run: |
+          npm run release
         env:
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
```

Comment:
Why do we need both?

Reply:
I unfortunately don't know why this is different in Node 15 (which seems to be the main difference between this repo and the other 4 vega repos we've set this up for. I tested with just one or the other, and it didn't work in either case. 

https://github.com/vega/vega-embed/commit/73e34deebd98013220f91f03f3a3f27be707cf7a